File: Capture the Flag/src/main/java/me/yoursole/ctf/DataFiles/Items/HermesBoots.java
package me.yoursole.ctf.DataFiles.Items;

import me.yoursole.ctf.DataFiles.GameData;
import me.yoursole.ctf.DataFiles.Utils;
import org.bukkit.*;
import org.bukkit.attribute.Attribute;
import org.bukkit.attribute.AttributeModifier;
import org.bukkit.enchantments.Enchantment;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerToggleFlightEvent;
import org.bukkit.inventory.ItemFlag;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.ShapedRecipe;
import org.bukkit.inventory.meta.Damageable;
import org.bukkit.inventory.meta.LeatherArmorMeta;

import java.util.ArrayList;
import java.util.List;

public class HermesBoots implements Listener {
    public static ItemStack item;

    static {
        ItemStack boots = new ItemStack(Material.LEATHER_BOOTS);
        LeatherArmorMeta meta = (LeatherArmorMeta)boots.getItemMeta();
        meta.setDisplayName(ChatColor.GRAY + "Hermes' Boots");

        List<String> lore = new ArrayList<>();
        lore.add(ChatColor.GOLD + "Item Ability: Messenger");
        lore.add(ChatColor.WHITE + "Allows you to double jump!");
        lore.add(ChatColor.WHITE + "Double Jumping damages the boots by 1 and the wearer by 1.");
        lore.add(ChatColor.BLUE + "Grants +25% Walk Speed when wearing.");
        meta.setLore(lore);
        meta.setColor(Color.fromRGB(0xb3acab));
        meta.addItemFlags(ItemFlag.HIDE_DYE, ItemFlag.HIDE_ATTRIBUTES);
        meta.addEnchant(Enchantment.DURABILITY, 10, true);
        meta.addAttributeModifier(Attribute.GENERIC_MOVEMENT_SPEED, new AttributeModifier("Hermes' Speed", 0.25, AttributeModifier.Operation.ADD_SCALAR));
        meta.addAttributeModifier(Attribute.GENERIC_ARMOR, new AttributeModifier("Hermes' Armor", 1, AttributeModifier.Operation.ADD_NUMBER));
        boots.setItemMeta(meta);
        HermesBoots.item = boots;

        NamespacedKey key = NamespacedKey.fromString("ctf:hermes");

        ShapedRecipe recipe = new ShapedRecipe(key, HermesBoots.item);
        recipe.shape("   ", "F F", "FDF");
        recipe.setIngredient('F', Material.FEATHER);
        recipe.setIngredient('D', Material.DIAMOND);
        recipe.setGroup("ctf:hermes");

        Bukkit.addRecipe(recipe);
        GameData.recipeKeys.put("hermes", key);
    }

    @EventHandler
    public void onAttemptFly(PlayerToggleFlightEvent event) {
        if (event.isFlying()) {
            Player player = event.getPlayer();
            ItemStack boots = event.getPlayer().getInventory().getArmorContents()[0];
            if (boots == null) return;
            if (Utils.compareBreakable(boots, HermesBoots.item)) {
                player.damage(1);
                Damageable meta = ((Damageable)boots.getItemMeta());
                meta.setDamage(meta.getDamage() + 1);
                if (meta.getDamage() > boots.getType().getMaxDurability()) {
                    player.getInventory().setBoots(null);
                }
                boots.setItemMeta(meta);

                player.setVelocity(player.getLocation().getDirection().multiply(1.2).setY(0.8));
                event.setCancelled(true);
                player.setAllowFlight(false);
                player.setFlying(false);
            }
        }
    }
}

File: src/main/kotlin/me/yoursole/ctf/datafiles/items/HermesBoots.kt
package me.yoursole.ctf.datafiles.items

import me.yoursole.ctf.datafiles.GameData
import me.yoursole.ctf.datafiles.Utils
import me.yoursole.ctf.datafiles.Utils.setCTFId
import org.bukkit.Bukkit
import org.bukkit.Color
import org.bukkit.Material
import org.bukkit.NamespacedKey
import org.bukkit.attribute.Attribute
import org.bukkit.attribute.AttributeModifier
import org.bukkit.craftbukkit.v1_18_R2.inventory.CraftItemStack
import org.bukkit.enchantments.Enchantment
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerToggleFlightEvent
import org.bukkit.inventory.EquipmentSlot
import org.bukkit.inventory.ItemFlag
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.ShapedRecipe
import org.bukkit.inventory.meta.Damageable
import org.bukkit.inventory.meta.LeatherArmorMeta
import java.util.*

object HermesBoots : Listener {
    val item: CraftItemStack by lazy {
        CraftItemStack.asCraftCopy(ItemStack(Material.LEATHER_BOOTS)).apply {
            itemMeta = (itemMeta as LeatherArmorMeta).apply {
                setDisplayName("§7Hermes' Boots")
                lore = listOf(
                    "§6Item Ability: Messenger",
                    "§fAllows you to double jump!",
                    "§fDouble Jumping damages the boots and the wearer by 1"
                )
                setColor(Color.fromRGB(0xb3acab))
                addItemFlags(ItemFlag.HIDE_DYE)
                addEnchant(Enchantment.DURABILITY, 10, true)
                addAttributeModifier(
                    Attribute.GENERIC_MOVEMENT_SPEED,
                    AttributeModifier(
                        UUID.randomUUID(),
                        "Hermes' Speed",
                        0.3,
                        AttributeModifier.Operation.ADD_SCALAR,
                        EquipmentSlot.FEET
                    )
                )
                addAttributeModifier(
                    Attribute.GENERIC_ARMOR,
                    AttributeModifier(
                        UUID.randomUUID(),
                        "Hermes' Armor",
                        1.25,
                        AttributeModifier.Operation.ADD_NUMBER,
                        EquipmentSlot.FEET
                    )
                )
            }
            setCTFId("hermes")
            val key = NamespacedKey.fromString("ctf:hermes")!!
            val recipe = ShapedRecipe(key, this)
            recipe.shape("   ", "F F", "FDF")
            recipe.setIngredient('F', Material.FEATHER)
            recipe.setIngredient('D', Material.DIAMOND)
            recipe.group = "ctf:hermes"
            Bukkit.addRecipe(recipe)
            GameData.recipeKeys["hermes"] = key
        }
    }

    @EventHandler
    fun onAttemptFly(event: PlayerToggleFlightEvent) {
        if (event.isFlying) {
            val player = event.player
            val boots = event.player.inventory.armorContents?.get(0)
            if (Utils.compareBreakable(boots, item)) {
                player.damage(1.0)
                val meta = boots?.itemMeta as Damageable
                meta.damage += 1
                if (meta.damage > boots.type.maxDurability) {
                    player.inventory.boots = null
                }
                boots.itemMeta = meta
                player.velocity = player.location.direction.multiply(1.25).setY(0.81)
                event.isCancelled = true
                player.allowFlight = false
                player.isFlying = false
            }
        }
    }
}